src/AppInstallerCLICore/Commands/SearchCommand.cpp

```diff
@@ -75,11 +76,26 @@ namespace AppInstaller::CLI
     {
         context.SetFlags(Execution::ContextFlag::TreatSourceFailuresAsWarning);
 
-        context <<
-            Workflow::OpenSource() <<
-            Workflow::SearchSourceForMany <<
-            Workflow::HandleSearchResultFailures <<
-            Workflow::EnsureMatchesFromSearchResult(false) <<
-            Workflow::ReportSearchResult;
+
+        if (context.Args.Contains(Execution::Args::Type::ListVersions))
+        {
+            context <<
+                Workflow::OpenSource() <<
+                Workflow::SearchSourceForMany <<

```

Comment:
Should this be SearchSourceForSingle?

Now I'm a little torn.

If we use `SearchSourceForMany`, it's highly likely search will return more than 1 result, and this will print "too many packages matched, please refine input" most of the times unless user inputs filters like --id --exact, etc.

If we use `SearchSourceForSingle`, the search will try to lock down on 1 result though other search candidates may be relevant. Sort of against the purpose of Search command.

Previously, Search command intends to be searching the broad and Show command intends to find 1 result. But with this feature, Search will behave differently based on whether --ListVersions are passed in. I somewhat feel --ListVersions does not relate well with purpose of Search.

@denelon, any thoughts?

Reply:
My thought was that it would be better to use `SearchSourceForMany` here, since as you said, filtering down is sort of against the purpose of the search command. I would expect that most users would be able to understand how to refine a search query down to a single package just by looking at the help page, especially when in many cases it will be as simple as adding `-e` or `-n 1`

As a user, I find the difference between `SearchSourceForMany` and `SearchSourceForSingle` quite confusing. Take a look at this for example -
```
PS C:\WINDOWS\system32> winget search resourcehacker
Name            Id                          Version
---------------------------------------------------
Resource Hacker AngusJohnson.ResourceHacker 5.1.8
PS C:\WINDOWS\system32> winget show resourcehacker
No package found matching input criteria.
```

Search clearly returned a single result, so why wouldn't show also find that single result? Is this actually a bug in `SearchSourceForSingle`? 

As to how this relates to the intent behind search being broad and show being for a single package, that is certainly a consideration and is actually the primary reason I left it as the `SearchSourceForMany`. More precisely, it was so that the search behavior was always consistent within the search command. The intent of this change is to reduce friction for users. As @denelon said in [#520](https://github.com/microsoft/winget-cli/issues/520#issuecomment-1251242279) -
>  I was looking at a YouTube video someone created for the Windows Package Manager. I noticed that they used "winget search" to try and find versions. I think this is a feature we want to implement. It's essentially adding "--versions" to the search command, and would produce the same output as "winget show --versions".

-----
The other option I thought of was, if there are multiple packages returned, to add a column to the table listing the latest few versions of each package. However, I decided against this due to scalability if more than just a few results are returned. Of course thresholds could be set, but that seemed excessive

Reply:
Regarding the ResourceHacker question, SearchForMany searches more fields with `substring` search, SearchForSingle on the other hand, searches only PackageName, Id, Moniker, PackageFamilyName and ProductCode with `CaseInsensitive` search, which is more strict and less likely to return multiple results.

Reply:
> Regarding the ResourceHacker question, SearchForMany searches more fields with `substring` search, SearchForSingle on the other hand, searches only PackageName, Id, Moniker, PackageFamilyName and ProductCode with `CaseInsensitive` search, which is more strict and less likely to return multiple results.

I see; Thank you for that clarification, that makes more sense.